Crash in evalTreeManagedResourceNoState

### Terraform Version

0.11.3, although I was using 0.11.1 when I initially produced this issue.

```
$ terraform -v
Terraform v0.11.3
+ provider.aws v1.6.0
```

### Terraform Configuration Files

```hcl
# Routed VPC

resource "aws_vpc" "ROUTED" {
  cidr_block = "${var.vpc_cidr_block}"
  instance_tenancy = "default"
  enable_dns_hostnames = "true"
  enable_dns_support = "true"
  enable_classiclink = "false"
  tags {
    Name = "ROUTED"
  }
}

# Public Subnets

resource "aws_subnet" "ROUTED-PUBLIC-A" {
  cidr_block = "${var.subnet_cidr_blocks["1A_Public"]}"
  vpc_id = "${aws_vpc.ROUTED.id}"
  map_public_ip_on_launch = "true"
  availability_zone = "${var.AWS_AZ["1A"]}"

  tags {
    Name = "ROUTED-PUBLIC-A"
  }
}

resource "aws_subnet" "ROUTED-PUBLIC-C" {
  cidr_block = "${var.subnet_cidr_blocks["1C_Public"]}"
  vpc_id = "${aws_vpc.ROUTED.id}"
  map_public_ip_on_launch = "true"
  availability_zone = "${var.AWS_AZ["1C"]}"

  tags {
    Name = "ROUTED-PUBLIC-C"
  }
}

# Private Subnets

resource "aws_subnet" "ROUTED-PRIVATE-A" {
  cidr_block = "${var.subnet_cidr_blocks["1A_Private"]}"
  vpc_id = "${aws_vpc.ROUTED.id}"
  map_public_ip_on_launch = "false"
  availability_zone = "${var.AWS_AZ["1A"]}"

  tags {
    Name = "ROUTED-PRIVATE-A"
  }
}

resource "aws_subnet" "ROUTED-PRIVATE-C" {
  cidr_block = "${var.subnet_cidr_blocks["1C_Private"]}"
  vpc_id = "${aws_vpc.ROUTED.id}"
  map_public_ip_on_launch = "false"
  availability_zone = "${var.AWS_AZ["1C"]}"

  tags {
    Name = "ROUTED-PRIVATE-C"
  }
}

# Internet GateWay

resource "aws_internet_gateway" "MAIN-ROUTED-GATEWAY" {
  vpc_id = "${aws_vpc.ROUTED.id}"

  tags {
    Name = "ROUTED"
  }
}

# Public Route table

resource "aws_route_table" "Public" {
  vpc_id = "${aws_vpc.ROUTED.id}"
  tags {
    Name = "Public"
  }
}

resource "aws_route" "public_default_vpc" {
  route_table_id = "${aws_route_table.Public.id}"
  destination_cidr_block = "172.31.0.0/16"
  vpc_peering_connection_id = "${aws_vpc_peering_connection.default_vpc.id}"
}

resource "aws_route" "public_internal_network" {
  route_table_id = "${aws_route_table.Public.id}"
  destination_cidr_block = "10.0.0.0/8"
  network_interface_id = "${var.vpn_interface_id}"
}

resource "aws_route" "public_internet" {
  route_table_id = "${aws_route_table.Public.id}"
  destination_cidr_block = "0.0.0.0/0"
  gateway_id = "${aws_internet_gateway.MAIN-ROUTED-GATEWAY.id}"
}


resource "aws_main_route_table_association" "Public" {
  route_table_id = "${aws_route_table.Public.id}"
  vpc_id = "${aws_vpc.ROUTED.id}"
}

# Private Route table:

resource "aws_route_table" "Private" {
  vpc_id = "${aws_vpc.ROUTED.id}"
  tags {
    Name = "Private"
  }
}

resource "aws_route" "private_default_vpc" {
  route_table_id = "${aws_route_table.Private.id}"
  destination_cidr_block = "172.31.0.0/16"
  vpc_peering_connection_id = "${aws_vpc_peering_connection.default_vpc.id}"
}

resource "aws_route" "private_internet" {
  route_table_id = "${aws_route_table.Private.id}"
  destination_cidr_block = "0.0.0.0/0"
  network_interface_id = "${var.vpn_interface_id}"
}

# Private Route Table assciation to Private subnets

resource "aws_route_table_association" "Private-A" {
  route_table_id = "${aws_route_table.Private.id}"
  subnet_id = "${aws_subnet.ROUTED-PRIVATE-A.id}"
}

resource "aws_route_table_association" "Private-C" {
  route_table_id = "${aws_route_table.Private.id}"
  subnet_id = "${aws_subnet.ROUTED-PRIVATE-C.id}"
}

# GET DEFAULT VPC ID

data "aws_vpc" "default_vpc" {
  filter {
    name = "tag:Name"
    values = ["DEFAULT"]
  }
}

output "default_vpc_id" {
  value = "${data.aws_vpc.default_vpc.id}"
}

# DHCP option resource

resource "aws_vpc_dhcp_options" "Routed" {
  domain_name = "${var.domain_name}"
  domain_name_servers = [
    "10.17.0.4",
    "10.17.64.4",
    "AmazonProvidedDNS"
  ]
  tags {
    Name = "ROUTED"
  }
}

# VPC peering connection resource

resource "aws_vpc_peering_connection" "default_vpc" {
  peer_vpc_id = "${data.aws_vpc.default_vpc.id}"
  vpc_id = "${aws_vpc.ROUTED.id}"
  accepter {
    allow_remote_vpc_dns_resolution = "true"
  }
  requester {
    allow_remote_vpc_dns_resolution = "true"
  }
  tags {
    Name = "legacy-vpc-bridge"
  }
}
```

### Crash Output
Backtrace:
```
$ terraform refresh
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xb4a081]

goroutine 2098 [running]:
github.com/hashicorp/terraform/terraform.(*NodeRefreshableManagedResourceInstance).evalTreeManagedResourceNoState(0xc42000ce48, 0xc4203dee70, 0xc42074e100)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/node_resource_refresh.go:219 +0x241
github.com/hashicorp/terraform/terraform.(*NodeRefreshableManagedResourceInstance).EvalTree(0xc42000ce48, 0x29a0240, 0xc42000ce48)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/node_resource_refresh.go:98 +0x71
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x29a0240, 0xc42000ce48, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/terraform/graph.go:113 +0xa3c
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc42037e690, 0x29a0240, 0xc42000ce48, 0xc4204241c0)
	/opt/gopath/src/github.com/hashicorp/terraform/dag/walk.go:387 +0x3c1
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/opt/gopath/src/github.com/hashicorp/terraform/dag/walk.go:310 +0x1364
```

Full crash log at https://gist.github.com/dbnicholson/78586259267afe3a439ac6926fa53a9a.

### Steps to Reproduce

Not really sure, but I had been importing and renaming state with `terraform init` and `terraform state mv`. This occurred after I finished and ran `terraform refresh`.

--- a/terraform/node_resource_refresh.go
+++ b/terraform/node_resource_refresh.go
@@ -213,10 +213,16 @@ func (n *NodeRefreshableManagedResourceInstance) evalTreeManagedResourceNoState(
 	// Determine the dependencies for the state.
 	stateDeps := n.StateReferences()
 
+	// n.Config can be nil if the config and state don't match
+	var raw *config.RawConfig
+	if n.Config != nil {
+		raw = n.Config.RawConfig.Copy()
+	}
+
 	return &EvalSequence{
 		Nodes: []EvalNode{
 			&EvalInterpolate{
-				Config:   n.Config.RawConfig.Copy(),
+				Config:   raw,
 				Resource: resource,
 				Output:   &resourceConfig,
 			},

